### PR TITLE
PODS-7668 event23

### PR DESCRIPTION
**Pull request by emrawsy**

### CODE REVIEW

File: test/forms/common/MembersDetailsFormProviderSpec.scala [hunk 1]
```diff
@@ -72,7 +72,7 @@ class MembersDetailsFormProviderSpec extends StringFieldBehaviours with Constrai
     behave like fieldThatBindsValidData(
       form,
       fieldName,
-      RegexpGen.from(regexSurname)
+      RegexpGen.from(regexName)
```
Comment (cyeniceri) [inline test/forms/common/MembersDetailsFormProviderSpec.scala:75]: same comments about the regex

File: test/forms/common/MembersDetailsFormProviderSpec.scala [hunk 2]
```diff
@@ -92,7 +92,7 @@ class MembersDetailsFormProviderSpec extends StringFieldBehaviours with Constrai
       form,
       fieldName,
       invalidString = "1A",
-      error = FormError(fieldName, invalidKey, Seq(regexSurname))
+      error = FormError(fieldName, invalidKey, Seq(regexName))
```
Comment (cyeniceri) [inline test/forms/common/MembersDetailsFormProviderSpec.scala:95]: same comments about the regex

File: test-utils/generators/UserAnswersGenerator.scala [hunk 3]
```diff
@@ -39,7 +41,7 @@ trait UserAnswersGenerator extends TryValues {
     arbitrary[(event1.SchemeUnAuthPaySurchargeMemberPage.type, JsValue)] ::
     arbitrary[(event1.ValueOfUnauthorisedPaymentPage.type, JsValue)] ::
     arbitrary[(event1.DoYouHoldSignedMandatePage.type, JsValue)] ::
-    arbitrary[(event1.MembersDetailsPage.type, JsValue)] ::
+//    arbitrary[(MembersDetailsPage.type, JsValue)] ::
```
Comment (cyeniceri) [inline test-utils/generators/UserAnswersGenerator.scala:44]: can we delete this if it is no longer used ?

**Review by cyeniceri — CHANGES_REQUESTED** [review]

File: app/models/enumeration/EventType.scala [hunk 4]
```diff
@@ -87,6 +87,8 @@ object EventType extends Enumerable.Implicits {
         stringBinder.unbind(key, value.toString)
     }
 
+  implicit val jsLiteral: JavascriptLiteral[EventType] = (value: EventType) => value.toString
```
Comment (cyeniceri) [inline app/models/enumeration/EventType.scala:90]: why do we need this implicit ?

**Review by cyeniceri — CHANGES_REQUESTED** [review]

File: app/models/enumeration/EventType.scala (hunk 4 above)
Comment (emrawsy) [reply app/models/enumeration/EventType.scala:90]: It was added to stop the event from being added to the URL in the page. The Address page has something similar.

File: app/forms/common/MembersDetailsFormProvider.scala [hunk 5]
```diff
@@ -37,7 +37,7 @@ class MembersDetailsFormProvider @Inject() extends Mappings with Transforms {
           text("membersDetails.error.lastName.required").verifying(
             firstError(
               maxLength(lastNameLength, "membersDetails.error.lastName.length"),
-              regexp(regexSurname, "membersDetails.error.lastName.invalid"))
+              regexp(regexName, "membersDetails.error.lastName.invalid"))
```
[the remark this answers is not in the document]
Comment (emrawsy) [reply app/forms/common/MembersDetailsFormProvider.scala:40]: deleted unused regex

File: conf/app.routes [hunk 6]
```diff
@@ -157,3 +155,14 @@ POST        /new-report/event-1-transfer-recipient                         contr
...
+
+GET         /new-report/event-23-entry-type                                controllers.event23.HowAddDualAllowanceController.onPageLoad(waypoints: Waypoints ?= EmptyWaypoints)
+POST        /new-report/event-23-entry-type                                controllers.event23.HowAddDualAllowanceController.onSubmit(waypoints: Waypoints ?= EmptyWaypoints)
+
+GET         /new-report/event-23-what-need                                 controllers.event23.WhatYouWillNeedController.onPageLoad(waypoints: Waypoints ?= EmptyWaypoints)
+
+GET         /new-report/event-23-member-details                            controllers.common.MembersDetailsController.onPageLoad(waypoints: Waypoints ?= EmptyWaypoints, eventType: EventType = Event23)
+POST        /new-report/event-23-member-details                            controllers.common.MembersDetailsController.onSubmit(waypoints: Waypoints ?= EmptyWaypoints, eventType: EventType = Event23)
+
+GET         /new-report/event-1-member-details                             controllers.common.MembersDetailsController.onPageLoad(waypoints: Waypoints ?= EmptyWaypoints, eventType: EventType = Event1)
```
[the remark this answers is not in the document]
Comment (emrawsy) [reply conf/app.routes]: added simpler routes.

**Review by cyeniceri — APPROVED** [review]

**Pull request merged**